Vic2ToHoI4/Source/V2World/Military/Army.h

```diff
@@ -19,6 +19,7 @@ class Army // also Navy
 	class Factory;
 	class Builder;
 
+	[[nodiscard]] const auto& getId() const { return id; }
```

Comment:
Add tests for this. I'd suggest one that shows the default (is a blank string what you want, for instance?), one that shows a basic id can be set, and one that demonstrates a failure of the regex match.